File: test7C#/test7/Model/Burger/VeggyBurger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace test7.Model.Burger
{
    class VeggyBurger : SimpleBurger
    {
        public VeggyBurger() : base("Veggy Burger", Convert.ToDecimal(6.99))
        {
            base.Filling(base.Filling() + ",Veggies");
        }
    }
}

File: test7Go/oop/models/burger/VeggyBurger.go
package burger

import (
	"fmt"
)

type VeggyBurger struct {
	SimpleBurger
}

func NewVeggyBurger() VeggyBurger {
	simpleBurger := NewSimpleBurger("Veggy Burger", 6.99)
	simpleBurger.Filling = fmt.Sprintf("%s,Veggies", simpleBurger.Filling)

	return VeggyBurger{simpleBurger}
}
